navigator_engine/pluggable_logic/conditional_functions.py

```diff
@@ -81,9 +81,34 @@ def check_spectrum_file(indicators: list[str], engine: DecisionEngine) -> bool:
                 f'Indicator "{indicator}" not found in Spectrum checklist'
             )
     indicator_checks = checklist[checklist['ID'].isin(indicators)]
-    indicator_checks['Status'] = indicator_checks['Status'].replace(
-        [0, 'FALSE', 'F', 'false', 'f'],
-        value=False,
-    )
+    indicator_checks['Status'].replace([0, 'FALSE', 'F', 'false', 'f'],
+                                       value=False,
+                                       inplace=True)
+
     indicator_checks['Status'][indicator_checks['Status'].ne(False)] = True
+
     return indicator_checks['Status'].eq(True).all()
+
+
+@register_conditional
+def check_naomi_file(indicators: list[str], engine: DecisionEngine) -> bool:
+    checklist = engine.data['naomi-validation-file']['data']
+
+    if checklist is None:
+        return False
+
+    for indicator in indicators:
+        if indicator not in checklist['NaomiCheckPermPrimKey'].values:
+            raise navigator_engine.common.DecisionError(
+                f'Indicator "{indicator}" not found in Naomi checklist'
+            )
+
+    indicator_checks = checklist[checklist['NaomiCheckPermPrimKey'].isin(indicators)]
+    indicator_checks['TrueFalse'].replace([0, 'FALSE', 'F', 'false', 'f'],
+                                          value=False,
+                                          inplace=True)
+
+    indicator_checks['TrueFalse'][indicator_checks['TrueFalse'].ne(False)] = True
+
+    return indicator_checks['TrueFalse'].eq(True).all()
```

Comment:
There's a lot of duplication here.  Can we create a single function `check_validation_file` taking the following arguments:
```
indicators  # = indicators
data_source  # = "naomi-validation-file"
result_column  # = "TrueFalse"
```

When you raise the decision errror your message could be:
```
raise navigator_engine.common.DecisionError(
    f'Indicator "{indicator}" not found in {data_source}'
)
```

You would then update:
```
@register_conditional
def check_naomi_file(indicators: list[str], engine: DecisionEngine) -> bool:
    return check_validation_file(
        indicators,
        'naomi-validation-file'
        'TrueFalse'
    )
```
And likewise for spectrum.